controllers/accounts_test.go

```diff
@@ -259,6 +269,183 @@ func (suite *AccountsTestSuite) TestAccountDeletionEndpointEnabled() {
 	suite.NotEqual(404, resp.Code)
 }
 
+func (suite *AccountsTestSuite) TestGet2FASettings() {
+	token, account := suite.createAuthSession()
+
+	// Test getting 2FA settings
+	req := httptest.NewRequest("GET", "/v2/accounts/2fa", nil)
+	req.Header.Set("Authorization", "Bearer "+token)
+	resp := util.ExecuteTestRequest(req, suite.router)
+	suite.Equal(http.StatusOK, resp.Code)
+
+	var settings datastore.TwoFADetails
+	util.DecodeJSONTestResponse(suite.T(), resp.Body, &settings)
+	suite.False(settings.TOTP)
+	suite.Nil(settings.TOTPEnabledAt)
+	suite.Nil(settings.RecoveryKeyCreatedAt)
+
+	// Enable 2FA
+	err := suite.ds.SetTOTPSetting(account.ID, true)
+	suite.Require().NoError(err)
+
+	// Test getting updated 2FA settings
+	req = httptest.NewRequest("GET", "/v2/accounts/2fa", nil)
+	req.Header.Set("Authorization", "Bearer "+token)
+	resp = util.ExecuteTestRequest(req, suite.router)
+	suite.Equal(http.StatusOK, resp.Code)
+
+	util.DecodeJSONTestResponse(suite.T(), resp.Body, &settings)
+	suite.True(settings.TOTP)
+	suite.NotNil(settings.TOTPEnabledAt)
+	suite.Nil(settings.RecoveryKeyCreatedAt)
+}
+
+func (suite *AccountsTestSuite) TestTOTPSetupAndFinalize() {
+	token, account := suite.createAuthSession()
+
+	// Test initializing TOTP setup
+	initReq := util.CreateJSONTestRequest("/v2/accounts/2fa/totp/init", controllers.TwoFAInitRequest{
+		GenerateQR: true,
+	})
+	initReq.Header.Set("Authorization", "Bearer "+token)
+	initResp := util.ExecuteTestRequest(initReq, suite.router)
+	suite.Equal(http.StatusOK, initResp.Code)
+
+	var initParsedResp controllers.TwoFAInitResponse
+	util.DecodeJSONTestResponse(suite.T(), initResp.Body, &initParsedResp)
+	suite.Require().NotEmpty(initParsedResp.URL)
+	suite.Require().NotNil(initParsedResp.QRCode)
+	suite.NotEmpty(*initParsedResp.QRCode)
+
+	// Parse the TOTP URL to extract the secret
+	key, err := otp.NewKeyFromURL(initParsedResp.URL)
+	suite.Require().NoError(err)
+	suite.Equal("Brave Account", key.Issuer())
```

Comment:
Let's check that the Algorithm is SHA-1 too, even though it's not part of the URL.